packages/server/src/adapters/aws-lambda.ts

```diff
@@ -0,0 +1,80 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { URLSearchParams } from 'url';
+import { resolveHTTPResponse } from '..';
+import { HTTPHeaders, HTTPRequest } from '../http/internals/types';
+import { AnyRouter, inferRouterContext } from '../router';
+import type { AWSLambdaOptions, LambdaContext } from './lambda-utils';
+
+export type { CreateLambdaContextOptions } from './lambda-utils';
+
+function lambdaEventToHTTPRequest(event: APIGatewayProxyEvent): HTTPRequest {
+  const query = new URLSearchParams();
+  for (const [key, value] of Object.entries(
+    event.multiValueQueryStringParameters ?? {},
+  )) {
+    for (const v of value ?? []) {
+      query.append(key, v);
+    }
+  }
+  return {
+    method: event.httpMethod,
+    query,
+    headers: event.headers,
+    body: event.body,
+  };
+}
+
+function transformHeaders(
+  headers: HTTPHeaders,
+): APIGatewayProxyResult['headers'] {
+  const obj: APIGatewayProxyResult['headers'] = {};
+
+  for (const [key, value] of Object.entries(headers)) {
+    if (typeof value === 'undefined') {
+      continue;
+    }
+    obj[key] = Array.isArray(value) ? value.join(', ') : value;
+  }
+  return obj;
+}
+
+export function createApiGatewayHandler<TRouter extends AnyRouter>(
+  opts: AWSLambdaOptions<TRouter>,
+): (
+  event: APIGatewayProxyEvent,
+  context: LambdaContext,
+) => Promise<APIGatewayProxyResult> {
+  return async (event: APIGatewayProxyEvent, context: LambdaContext) => {
+    const req = lambdaEventToHTTPRequest(event);
+    const path = event.path;
+
+    const createContext = async function _createContext(): Promise<
+      inferRouterContext<TRouter>
+    > {
+      return await opts.createContext?.({ event, context });
+    };
+
+    const response = await resolveHTTPResponse({
+      router: opts.router,
+      batching: opts.batching,
+      responseMeta: opts?.responseMeta,
+      createContext,
+      req,
+      path,
+      error: null,
+      onError(o) {
+        opts?.onError?.({
+          ...o,
+          req: event,
+        });
+      },
+    });
+
+    const resp: APIGatewayProxyResult = {
+      statusCode: response.status,
+      body: response.body ?? '',
```

Comment:
I don't think it makes sense for this to be an empty string. If we're using payload format v2, [body can be `undefined`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/05ca376629ce89983fb50d6af08c85d81a111734/types/aws-lambda/trigger/api-gateway-proxy.d.ts#L236).